assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,171 @@
-console.log('***** Music Collection *****')
+console.log("***** Music Collection *****");
+// ---------------
+// Data
+// ---------------
+import { tracksData, testTracks, testTracks7 } from "../utils/tracksData.js";
+import { formatDuration } from "../utils/formatDuration.js";
+
+// collection array
+let collection = [];
+
+// ---------------
+// Functions
+// ---------------
+
+// add to collection function
+const addToCollection = (title, artist, yearPublished, tracks) => {
+  const addition = { title, artist, yearPublished, tracks };
+  collection.push(addition);
+  return addition;
+};
+
+// show collection function
+const showCollection = (array) => {
+  console.log("number of items to show", array.length);
+  for (let album of array) {
+    const { title, artist, yearPublished } = album;
+    console.log(`${title} by ${artist}, published in ${yearPublished}`);
+    for (let i = 0; i < album.tracks.length; i++) {
+      console.log(
+        `${i + 1}. ${album.tracks[i].name}: ${formatDuration(
+          album.tracks[i].duration
+        )}`
+      );
+    }
+  }
+};
+
+// fing by artist function
+const findByArtist = (artist) => {
+  let results = [];
+  for (let album of collection) {
+    if (album.artist.toLowerCase() === artist.toLowerCase()) {
+      results.push(album);
+    }
+  }
+
+  return results;
+};
+
+const search = ({ artist, year, trackName = "" } = {}) => {
+  if (trackName) {
+    const results = collection.filter((album) => {
+      for (let track of album.tracks) {
+        if (track.name.toLowerCase() === trackName.toLowerCase()) {
+          return true;
+        }
+      }
+      return false;
+    });
+    return results;
+  }
+  if (!artist || !year) {
+    return collection;
+  }
+  const results = collection.filter((album) => {
```

Comment:
Great use of filter. I would go a step further and just `return collection.filter(...)` directly with no `results variable.